File: CSharp/HtmlDocToolTest/CleanerTest.cs
using Html5DomWalker;
using HtmlDocCleaner;
using HtmlTest.Base;
using HtmlWalker.Platform;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace HtmlDocToolTest
{
    [TestClass]
    public class CleanerTest : BaseTest
    {
        public CleanerTest() : base("Cleaner") { }

        public CleanerTest(string subFolder) : base(subFolder) { }

        protected override WalkerPlatform Platform => Html5DomPlatform.Instance;

        protected override void CreateExpected(FileInfo expected, FileInfo testInput, TestOptions options)
            => RunProgram(testInput, expected, options);

        private void RunProgram(FileInfo testInput, FileInfo testOutput, TestOptions options)
        {
            var args = new List<string>() { testInput.FullName, testOutput.FullName };
            if (options.DocumentOptions.Encoding != null)
            {
                args.Add("--input-encoding");
                args.Add(options.DocumentOptions.Encoding.CodePage.ToString());
            }
            Program.Main(args.ToArray());
        }

        public void DoTest(FileInfo testInput, FileInfo testOutput, FileInfo expected, TestOptions options)
        {
            RunProgram(testInput, testOutput, options);
            CompareToExpected(testOutput, expected, false);
        }

        protected void DoTest(string test)
        {
            var options = new TestOptions(Platform) { AutoCreate = false };
            var testInput = GetTestInput("test." + test + ".html");
            var expected = GetTestExpected("saved." + test + ".html", testInput, options);
            var testOutput = GetTestOutput("output." + test + ".html");
            DoTest(testInput, testOutput, expected, options);
        }

        protected void DoWordTest(string test)
        {
            var options = new TestOptions(Platform) { AutoCreate = false };
            options.DocumentOptions.Encoding = CodePagesEncodingProvider.Instance.GetEncoding(1252);

            var testInput = GetTestInput("test." + test + ".html");
            var expected = GetTestExpected("saved." + test + ".html", testInput, options);
            var testOutput = GetTestOutput("output." + test + ".html");
            DoTest(testInput, testOutput, expected, options);
        }

        [TestMethod]
        [TestCategory("Html5")]
        public void Comment_01() => DoTest("comment.01");

        [TestMethod]
        [TestCategory("Html5")]
        public void Head_01() => DoTest("head.01");

        [TestMethod]
        [TestCategory("Html5")]
        public void Div_01() => DoTest("div.01");

        [TestMethod]
        [TestCategory("Html5")]
        public void Attribute_01() => DoTest("attribute.01");

        [TestMethod]
        [TestCategory("Word")]
        [Description("One line paragrach")]
        public void Word_01() => DoWordTest("word.01");

        [TestMethod]
        [TestCategory("Word")]
        [Description("Bullet list")]
        public void Word_02() => DoWordTest("word.02");
    }
}

File: Java/htmlwalker/src/test/java/htmltest/htmldoctooltest/CleanerTest.java
package htmltest.htmldoctooltest;

import java.io.File;

import org.junit.Test;

import htmldoccleaner.App;
import htmljsoupwalker.HtmlPlatform;
import htmltest.base.BaseTest;
import htmlwalker.platform.WalkerPlatform;

public class CleanerTest extends BaseTest
{
	private final static String TSTPATH = "Test";

    public CleanerTest() { super("Cleaner"); }

    @Override
    protected String testFolder() { return TSTPATH; }

    @Override
    protected String outFolder() { return TSTPATH; }

    @Override
    protected WalkerPlatform platform() { return HtmlPlatform.theInstance; }

    @Override
    protected void createExpected(File expected, File testInput, TestOptions options)
    {
        App.main(new String[] { testInput.getPath(), expected.getPath() });
    }

    protected void doTest(String test)
    {
        var options = new TestOptions(platform());
        options.autoCreate = false;

        var testInput = getTestInput("test." + test + ".html");
        var expected = getTestExpected("saved." + test + ".html", testInput, options);
        var testOutput = getTestOutput("output." + test + ".html");
        App.main(new String[] { testInput.getPath(), testOutput.getPath() });
        compareToExpected(testOutput, expected, false);
    }

    @Test
    public void comment_01() { doTest("comment.01"); }

    @Test
    public void head_01() { doTest("head.01"); }

    @Test
    public void div_01() { doTest("div.01"); }

    @Test
    public void attribute_01() { doTest("attribute.01"); }

    @Test
    public void word_01() { doTest("word.01"); }
}
